src/contracts/core/Slasher.sol

```diff
@@ -1,102 +1,368 @@
+     * @param strategy the strategy to get the pending slashing rate for
+     * @param operatorSet the operator set to get the pending slashing rate for
+     *
+     * @return the parts per hundred million that will be slashed for the given
+     * operator, strategy, epoch, and operator set assuming no further
+     * modifications to requested slashing rate by operatorSet
+     */
+    function getPendingSlashingRate(
+        address operator,
+        IStrategy strategy,
+        IOperatorSetManager.OperatorSet calldata operatorSet,
+        uint32 epoch
+    ) external view returns (uint32) {
+        uint32 requestedSlashingRate = getRequestedSlashingRate(operator, strategy, operatorSet, epoch);
+        if (requestedSlashingRate >= SlashingAccountingUtils.BIPS_FACTOR_SQUARED) {
+            requestedSlashingRate = uint32(SlashingAccountingUtils.BIPS_FACTOR_SQUARED);
+        }
+        return requestedSlashingRate;
+    }
 
-    function delegation() external view returns (IDelegationManager) {}
+    /**
+     * @notice fetches the parts per hundred million that will be slashed for
+     * the given operator, strategy, and epoch, across all operator set assuming
+     * no more modifications to requested slashing rate for the operator.
+     *
+     * @param operator the operator to get the pending slashing rate for
+     * @param strategy the strategy to get the pending slashing rate for
+     * @param epoch the epoch to get the pending slashing rate for
+     *
+     * @return the parts per hundred million that will be slashed for the
+     * given operator, strategy, and epoch, across all operator set assuming
+     * no more modifications to requested slashing rate for the operator.
+     */
+    function getTotalPendingSlashingRate(
+        address operator,
+        IStrategy strategy,
+        uint32 epoch
+    ) external view returns (uint32) {
+        uint64 totalSlashingRate = slashingRequests[operator][strategy][epoch].slashingRate;
+        if (totalSlashingRate >= SlashingAccountingUtils.BIPS_FACTOR_SQUARED) {
+            totalSlashingRate = SlashingAccountingUtils.BIPS_FACTOR_SQUARED;
+        }
+        return uint32(totalSlashingRate);
+    }
 
-    function isFrozen(address) external view returns (bool) {}
+    /**
+     * @notice gets whether withdrawals of the given strategy delegated to the given operator can be withdrawn and the scaling factor
+     * @param operator the operator the withdrawal is delegated to
+     * @param strategy the strategy the withdrawal is from
+     * @param epoch the last epoch the withdrawal was slashable until
+     * @return whether the withdrawal can be executed
+     * @return whether there was a slashing request for the given operator and strategy at the given epoch
+     */
+    function getWithdrawabilityAndScalingFactorAtEpoch(
+        address operator,
+        IStrategy strategy,
+        uint32 epoch
+    ) external view returns (bool, uint64) {
```

Comment:
Mega wordy. Suggestions:
- `getWithdrawability`
- `getScalingFactor`

Don't need to include the full depth and breadth of function in the name. Good documentation will make these pop out -- currently you're forcing this to be called as follows:

```solidity
(bool canWithdraw, uint64 scalingFactor) =
  slasher.getWithdrawabilityAndScalingFactorAtEpoch(operator, strategy, epoch);
```

Naming the return variables is already going to make the usage apparent. It could be super simple instead:

```solidity
(bool canWithdraw, uint64 scalingFactor) = 
  slasher.getWithdrawability(operator, strategy, epoch);
```

Reply:
See #609